tests/Unit/controllers/account.test.js

```diff
@@ -0,0 +1,494 @@
+      const expectedAccountId = "asd2123";
+
+      const expectedAccount = {
+        userName: "test",
+        email: "test@test.com",
+        expectedDiscrminator,
+        _id: expectedAccountId,
+      };
+
+      const req = {
+        body: { ...expectedAccount },
+      };
+
+      const res = {
+        status: jest.fn((status) => {
+          return res;
+        }),
+        json: jest.fn((json) => {
+          return res;
+        }),
+      };
+
+      const next = jest.fn();
+
+      accountRepository.create = jest.fn((opts) => {
+        const { userName, email } = opts.newAccount;
+
+        return {
+          userName,
+          email,
+          expectedDiscrminator,
+          _id: expectedAccountId,
+        };
+      });
+
+      accountRepository.generateAuthToken = jest.fn((account) => expectedToken);
+
+      relationshipsRepository.create = jest.fn((account) => {});
+
+      groupsRepository.create = jest.fn((account) => {});
+
+      await controller.create(req, res, next);
+
+      expect(accountRepository.create).toHaveBeenCalledTimes(1);
+      expect(accountRepository.create).toBeCalledWith({
+        newAccount: expectedAccount,
+      });
+      expect(relationshipsRepository.create).toHaveBeenCalledTimes(1);
+      expect(relationshipsRepository.create).toBeCalledWith(expectedAccountId);
+      expect(groupsRepository.create).toHaveBeenCalledTimes(1);
+      expect(groupsRepository.create).toBeCalledWith(expectedAccountId);
+      expect(accountRepository.generateAuthToken).toHaveBeenCalledTimes(1);
+      expect(accountRepository.generateAuthToken).toBeCalledWith(
+        expectedAccount
+      );
+      expect(res.status).toHaveBeenCalledTimes(1);
+      expect(res.status).toBeCalledWith(httpStatus.CREATED);
+      expect(res.json).toHaveBeenCalledTimes(1);
+      expect(res.json).toBeCalledWith({ token: expectedToken });
+      expect(next).toHaveBeenCalledTimes(0);
+    });
+
+    test("Should throw invalid object error when request contains invalid falids and handle it", async () => {
+      //This test will fail until we redo validation
+      const expectedError = new Error(errors.INVALID_OBJECT);
+      expectedError.name = errors.INVALID_OBJECT;
+
+      const req = { body: {} };
+      const res = {};
+      const next = jest.fn();
+
+      await controller.create(req, res, next);
+
+      expect(next).toHaveBeenCalledTimes(1);
+      expect(next).toBeCalledWith(expectedError);
+    });
+
+    test("Should handle errors", async () => {
+      const expectedError = new Error(errors.UNKNOWN);
+      expectedError.name = errors.UNKNOWN;
+
+      accountRepository.create = jest.fn((opts) => {
+        throw expectedError;
+      });
+
+      const req = { body: {} };
+      const res = {};
+      const next = jest.fn();
+
+      await controller.create(req, res, next);
+
+      expect(next).toHaveBeenCalledTimes(1);
+      expect(next).toBeCalledWith(expectedError);
```

Comment:
wouldnt this line implies that the previous line have passed? So the previous line is redundant

Reply:
No this is expecting to be called specifically with this value. The function does not actually care what you pass to it otherwise. Even if the previously passes does not imply this line will pass.

Reply:
Hmm never mind. I see what you mean. I'll change that.

Reply:
there are many instances of this though, be sure to change them too

Reply:
Some of them are on purpose